File: app/src/main/java/knf/kuma/recents/RecentsViewModel.java
package knf.kuma.recents;

import android.content.Context;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;
import knf.kuma.database.CacheDB;
import knf.kuma.pojos.RecentObject;
import knf.kuma.retrofit.Repository;

public class RecentsViewModel extends ViewModel {
    private Repository repository=new Repository();

    public RecentsViewModel() {
    }

    public void reload(Context context) {
        repository.reloadRecents(context);
    }

    public LiveData<List<RecentObject>> getDBLiveData(){return CacheDB.INSTANCE.recentsDAO().getObjects();}
}

File: app/src/main/java/knf/kuma/recents/RecentsViewModel.kt
package knf.kuma.recents

import androidx.lifecycle.ViewModel
import knf.kuma.database.CacheDB
import knf.kuma.pojos.av1.RecentAV1
import knf.kuma.retrofit.Repository
import kotlinx.coroutines.flow.Flow

class RecentsViewModel : ViewModel() {
    private val repository = Repository()

    val dbFlowData: Flow<List<RecentAV1>>
        get() = CacheDB.INSTANCE.recentAV1DAO().allFlow

    fun reload() {
        repository.reloadRecents()
    }
}
